src/components/PostsList.tsx

```diff
@@ -1,56 +1,61 @@
 import React from 'react';
-import { Post } from '../types/Post';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, max-len
+import * as selectedPostActions from '../features/selectedPost/selectedPostSlice';
 
 type Props = {
-  posts: Post[],
   selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
 };
 
 export const PostsList: React.FC<Props> = ({
-  posts,
   selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+}) => {
+  const dispatch = useAppDispatch();
+  const { items } = useAppSelector(state => state.posts);
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-      <tbody>
-        {posts.map(post => (
-          <tr key={post.id} data-cy="Post">
-            <td data-cy="PostId">{post.id}</td>
-            <td data-cy="PostTitle">{post.title}</td>
-            <td className="has-text-right is-vcentered">
-              <button
-                type="button"
-                data-cy="PostButton"
-                className={classNames(
-                  'button',
-                  'is-link',
-                  {
-                    'is-light': post.id !== selectedPostId,
-                  },
-                )}
-                onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
-                }}
-              >
-                {post.id === selectedPostId ? 'Close' : 'Open'}
-              </button>
-            </td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
           </tr>
-        ))}
-      </tbody>
-    </table>
-  </div>
-);
+        </thead>
+
+        <tbody>
+          {items.map(post => (
+            <tr key={post.id} data-cy="Post">
+              <td data-cy="PostId">{post.id}</td>
+              <td data-cy="PostTitle">{post.title}</td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  type="button"
+                  data-cy="PostButton"
+                  className={classNames(
+                    'button',
+                    'is-link',
+                    {
+                      'is-light': post.id !== selectedPostId,
+                    },
+                  )}
+                  onClick={() => {
+                    dispatch(selectedPostActions.add(
+                      (post.id === selectedPostId ? null : post),
+                    ));
+                  }}
```

Comment:
consider to make variable `const isSelected = post.id === selectedPostId`  and use in all places
```suggestion
                  onClick={onAdd}
```

it is bad practise to make anonymous arrow functions in jsx. instead make handler in component and pass reference here.

```
cosnt onAdd = () => dispatch(
  selectedPostActions.add(isSelected ? null : post )
);
```
check all similar cases